pkg/monitor/monitor/monitor_manager.go

```diff
@@ -272,3 +285,44 @@ func (mm *MonitorManager) syncTidbMonitorRbac(monitor *v1alpha1.TidbMonitor) (*c
 
 	return sa, nil
 }
+
+func (mm *MonitorManager) syncIngress(monitor *v1alpha1.TidbMonitor) error {
+	if err := mm.syncPrometheusIngress(monitor); err != nil {
+		return err
+	}
+
+	return mm.syncGrafanaIngress(monitor)
+}
+
+func (mm *MonitorManager) syncPrometheusIngress(monitor *v1alpha1.TidbMonitor) error {
+	if monitor.Spec.Prometheus.Ingress == nil {
+		return mm.removeIngressIfExist(monitor, prometheusName(monitor))
+	}
```

Comment:
should we check `monitor.Spec.Prometheus` too like grafana?

Reply:
We don't have to check as `Prometheus` is a non-nil in `TidbMonitorSpec`